lib/src/database.dart

```diff
@@ -152,6 +164,47 @@ class Query {
         .then((snapshot) => new DataSnapshot(snapshot));
   }
 
+  /// Detaches a callback previously attached with [on].
+  ///
+  /// Note that if [on] was called multiple times with the same [eventType] and
+  /// [callback], the callback will be called multiple times for each event, and
+  /// [off] must be called multiple times to remove the callback. Calling [off]
+  /// on a parent listener will not automatically remove listeners registered on
+  /// child nodes, [off] must also be called on any child listeners to remove
+  /// the callback.
+  ///
+  /// NOTE different from JS API !
+  /// If [eventType] is specifiede, all callbacks for that specified
+  /// [eventType] will be removed. If no [eventType] is
+  /// specified, all callbacks for the [Reference] will be removed.
+  /// To unsubscribe a specific callback, use [QuerSubscription.cancel] method
+  void off([String eventType]){
+    if (eventType != null){
+      nativeInstance.off(eventType);
+    }
+    else {
+      nativeInstance.off();
+    }
+  }
+
+  /// Listens for data changes at a particular location.
+  ///
+  /// This is the primary way to read data from a [Database]. Your callback will
+  /// be triggered for the initial data and again whenever the data changes.
+  /// Use [off] to stop receiving updates.
+  ///
+  /// return function to unsubscribe
+  QuerySubscription on<T>(
+      String eventType, Function(DataSnapshot<T> snapshot) callback,
+      [Function() cancelCallback]) {
+    var fn = allowInterop((snapshot) => callback(new DataSnapshot(snapshot)));
+    if (cancelCallback != null)
+      nativeInstance.on(eventType, fn, allowInterop(cancelCallback));
+    else
+      nativeInstance.on(eventType, fn);
```

Comment:
Please add curly braces for `if else` here.
Effective dart: https://www.dartlang.org/guides/language/effective-dart/style#do-use-curly-braces-for-all-flow-control-structures